--- android_icu4j/src/main/java/android/icu/text/BreakIterator.java	2019-01-30 16:31:03.631387738 +0000
+++ android_icu4j/src/main/java/android/icu/text/BreakIterator.java	2019-01-30 16:29:57.372107027 +0000
@@ -543,7 +543,7 @@
     public static final int KIND_SENTENCE = 3;
     /**
      * <strong>[icu]</strong>
-     * @deprecated on Android but not deprecated in ICU
+     * @deprecated Use {@link #KIND_WORD} instead.
      */
     @Deprecated
     public static final int KIND_TITLE = 4;
